### PR TITLE
Allow moving bots between channels.

### PR DESCRIPTION
**Pull request by efindus**
Currently when bot is moved from one channel to another player disconnects. This fixes the issue.

### CODE REVIEW

File: src/lib/Manager.ts [hunk 1]
```diff
@@ -162,7 +162,18 @@ export class Manager extends EventEmitter {
     public voiceStateUpdate(data: VoiceStateUpdate): Promise<boolean> {
         if (data.user_id !== this.user) return Promise.resolve(false);
 
-        if (data.channel_id) {
+        if (data.channel_id && data.guild_id) {
+            if (this !== undefined) {
```
Comment (MrJacz) [inline src/lib/Manager.ts]: remove this

File: src/lib/Manager.ts [hunk 2]
```diff
@@ -162,7 +162,18 @@ export class Manager extends EventEmitter {
...
+                if (this.voiceStates !== undefined) {
```
Comment (MrJacz) [inline src/lib/Manager.ts]: remove this

File: src/lib/Manager.ts [hunk 3]
```diff
@@ -162,7 +162,18 @@ export class Manager extends EventEmitter {
...
+                    if (this.voiceStates.has(data.guild_id) !== false) {
```
Comment (MrJacz) [inline src/lib/Manager.ts]:
```suggestion
                    if (this.voiceStates.has(data.guild_id)) {
```

File: src/lib/Manager.ts [hunk 4]
```diff
@@ -162,7 +162,18 @@ export class Manager extends EventEmitter {
...
+                        if (this.voiceStates.get(data.guild_id).channel_id !== undefined) {
```
Comment (MrJacz) [inline src/lib/Manager.ts]:
```suggestion
                        if (this.voiceStates.get(data.guild_id)!.channel_id) {
```

File: src/lib/Manager.ts [hunk 5]
```diff
@@ -162,7 +162,18 @@ export class Manager extends EventEmitter {
...
+                            if (data.channel_id !== this.voiceStates.get(data.guild_id).channel_id) {
```
Comment (MrJacz) [inline src/lib/Manager.ts]: Make the channelID a variable instead of using getting it twice

**Review by MrJacz — CHANGES_REQUESTED** [review]

**Comment by efindus** [comment]
Yes! Finally!

**Comment by MrJacz** [comment]
Now that I look at the code more and more, I was just thinking, If you remove the entire `if (data.channel_id) block does that work properly?

**Comment by efindus** [comment]
```js
this.voiceServers.delete(data.guild_id);
this.voiceStates.delete(data.guild_id);
return Promise.resolve(false);
```

But what with this then? When change is about bot, it will always delete the data, meaning, when bot joins channel to play something, it deletes data immediately after, so nothing is being played and bot stays idle.

**Pull request closed without merging**